Running and network splitting, and the Introspector app

ktoso opened the pull request:
Almost there... Took me quite a while :?
Should have some results before standup I hope though.

patriknw commented on multidc/network-fleaky-node.sh:
```diff
@@ -0,0 +1,32 @@
+all=( "$nodeF_1" "$nodeF_2" "$nodeF_3" "$nodeI_1" "$nodeI_2" "$nodeI_3" )
+
+for node in "${all[@]}"; do
+  echo "============ Starging MultiDC App on:       $node     ... ============"
```
Staging

patriknw commented on multidc/network-split.sh:
```diff
@@ -0,0 +1,22 @@
+    echo "============ Separating Frankfurt node:       $node     ... ============"
+
+    for separate_from_node in "${nodes_west_ip}"; do
+      command="sudo iptables -A INPUT -p udp -s $separate_from_node -j DROP"
```
we probably want to slit the Cassandra cluster also, which is using tcp, I assume

patriknw replied:
are we doing bidirectional split, or is this just dropping in one direction?
I think weird things will happen if you only drop in one direction.

ktoso replied:
That's one direction, but we can easily write complete splits with this, loop through other side and do the same

patriknw replied:
When I did such things I had trouble if it took too long between the splits in the different directions, e.g. first looping 30 nodes and splitting in one direction and then splitting 30 in other direction can be problematic.
Better to execute the 2 ssh commands immediately after each other.

ktoso replied:
I see, we can launch all these with `&` then i guess, will think about timing

patriknw commented on multidc/run-multi-dc-test-remote.sh:
```diff
@@ -0,0 +1,45 @@
+. nodes-bash-exports.sh
+
+for node in "${nodes_central[@]}"; do
+  echo "============ Starging MultiDC App on:       $node     ... ============"
```
Staging

ktoso commented:
Updated the PR.

Workflow is basically:

- start all nodes
- make sure to have `replicated-entity.pem`, it's in the team `keybase`
- go to `infra`
- `make-re-node-files-from-aws.sh` -- refreshes all the IPs and seed nodes in local files
- `fix` all nodes
  - `fix nodes_with_role:cassandra-re` -- actually updates the servers with the above
  - `fix nodes_with_role:akka-re` -- actually updates the servers with the above
- all this was from `infra`, now move back to `multidc` and from here you can develop and experiment
- run the app on all nodes `run-multi-dc-test-remote.sh`
  - IPs are automatically obtained from `nodes-bash-exports.sh`, though it should be done a bit nicer perhaps later on with using `aws` command line
- if you want to experiment, use `rsync-local-src-to-akka-nodes.sh` to sync the sources (`src`) to all nodes
  - then kill all nodes `kill-multi-dc-test-remote-run.sh` and start them again
- if you want to introduce package drops use `network-fleaky-node.sh`, which is not complete yet
- `network-split.sh` is used for making a split; do look into the code and edit to get exaclty what you want. Syntax is `network-split.sh split` or `network-split.sh heal`

ktoso commented:
HTTP ports are open so you can query and drive the tests from your local box by sending `$ http :19999/...` (`/members` for overview etc, see README.md)

ktoso commented on multidc/src/main/resources/logback.xml:
```diff
@@ -1,20 +1,35 @@
+    </root>
+    <logger level="INFO">
+        <appender-ref ref="FILE"/>
+    </logger>
```
does this make sense actually or there can only be one?

ktoso replied:
fixed

ktoso commented:
Where I got to basically:

```

[info] 16:51:20.603 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,a))
[info] 16:51:20.603 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603))
[info] 16:51:20.603 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,a))
[info] 16:51:20.603 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603))
[info] 16:51:20.604 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,b))
[info] 16:51:20.604 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603))
[info] 16:51:20.604 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,c))
[info] 16:51:20.604 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603))
[info] 16:51:20.604 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,d))
[info] 16:51:20.605 [MultiDcSystem-akka.actor.default-dispatcher-16] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603))
[info] 16:51:20.605 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,d))
[info] 16:51:20.605 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604))
[info] 16:51:20.605 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,d))
[info] 16:51:20.605 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604))
[info] 16:51:20.606 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,d))
[info] 16:51:20.606 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604))
[info] 16:51:20.606 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,d))
[info] 16:51:20.606 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604))
[info] 16:51:20.606 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,d))
[info] 16:51:20.606 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604))
[info] 16:51:20.611 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Recovery complete: Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604))
[info] 16:51:20.622 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Inspect arrived; all state: AllState(List(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604)))
[info] 16:51:28.996 [MultiDcSystem-akka.actor.default-dispatcher-18] INFO  kappa(akka://MultiDcSystem) - Append(kappa,e) arrived;
[info] 16:51:28.997 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,e))
[info] 16:51:28.997 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997))
[info] 16:51:29.027 [MultiDcSystem-akka.actor.default-dispatcher-9] WARN  c.l.m.ReplicatedIntrospector - You should start Sharding proxy for entity type [introspector] to data center [eu-central], otherwise the speculative replication optimization is disabled. You should also handle ReplicatedEntity.SpeculativeReplicatedEvent message in sharding entityId and shardId message extractors.
[info] 16:51:29.028 [MultiDcSystem-akka.actor.default-dispatcher-9] INFO  kappa(akka://MultiDcSystem) -      sending: AllState(List(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997)))
[info] 17:13:47.854 [MultiDcSystem-akka.actor.default-dispatcher-9] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@18.194.243.77:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@34.240.43.90:2551, dataCenter = eu-west, status = Up), Member(address = akka://MultiDcSystem@54.154.72.165:2551, dataCenter = eu-west, status = Up)]. Node roles [dc-eu-central]
[info] 17:13:48.110 [MultiDcSystem-akka.actor.default-dispatcher-3] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@34.240.43.90:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@18.194.243.77:2551, dataCenter = eu-central, status = Up), Member(address = akka://MultiDcSystem@52.59.221.199:2551, dataCenter = eu-central, status = Up), Member(address = akka://MultiDcSystem@52.59.241.114:2551, dataCenter = eu-central, status = Up)]. Node roles [dc-eu-west]
[info] 17:13:48.254 [MultiDcSystem-akka.actor.default-dispatcher-12] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@34.253.229.136:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@52.59.221.199:2551, dataCenter = eu-central, status = Up), Member(address = akka://MultiDcSystem@52.59.241.114:2551, dataCenter = eu-central, status = Up)]. Node roles [dc-eu-west]
[info] 17:13:48.247 [MultiDcSystem-akka.actor.default-dispatcher-4] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@52.59.221.199:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@34.240.43.90:2551, dataCenter = eu-west, status = Up), Member(address = akka://MultiDcSystem@34.253.229.136:2551, dataCenter = eu-west, status = Up), Member(address = akka://MultiDcSystem@54.154.72.165:2551, dataCenter = eu-west, status = Up)]. Node roles [dc-eu-central]
[info] 17:13:48.623 [MultiDcSystem-akka.actor.default-dispatcher-18] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@52.59.241.114:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@34.240.43.90:2551, dataCenter = eu-west, status = Up), Member(address = akka://MultiDcSystem@34.253.229.136:2551, dataCenter = eu-west, status = Up)]. Node roles [dc-eu-central]
[info] 17:13:49.172 [MultiDcSystem-akka.actor.default-dispatcher-5] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@54.154.72.165:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@18.194.243.77:2551, dataCenter = eu-central, status = Up), Member(address = akka://MultiDcSystem@52.59.221.199:2551, dataCenter = eu-central, status = Up)]. Node roles [dc-eu-west]
[info] 17:14:27.169 [MultiDcSystem-akka.actor.default-dispatcher-5] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@54.154.72.165:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@52.59.241.114:2551, dataCenter = eu-central, status = Up)]. Node roles [dc-eu-west]
[info] 17:14:27.621 [MultiDcSystem-akka.actor.default-dispatcher-4] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@52.59.241.114:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@54.154.72.165:2551, dataCenter = eu-west, status = Up)]. Node roles [dc-eu-central]
[info] 17:14:31.852 [MultiDcSystem-akka.actor.default-dispatcher-2] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@18.194.243.77:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@34.253.229.136:2551, dataCenter = eu-west, status = Up)]. Node roles [dc-eu-central]
[info] 17:14:32.253 [MultiDcSystem-akka.actor.default-dispatcher-15] WARN  a.c.ClusterCoreDaemon - Cluster Node [akka://MultiDcSystem@34.253.229.136:2551] - Marking node(s) as UNREACHABLE [Member(address = akka://MultiDcSystem@18.194.243.77:2551, dataCenter = eu-central, status = Up)]. Node roles [dc-eu-west]
[info] 17:14:38.965 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Append(kappa,split-a) arrived;
[info] 17:14:38.966 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,split-a))
[info] 17:14:38.969 [MultiDcSystem-akka.actor.default-dispatcher-9] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965))
[info] 17:14:38.973 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      sending: AllState(List(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965)))
[info] 17:14:42.635 [MultiDcSystem-akka.actor.default-dispatcher-17] INFO  kappa(akka://MultiDcSystem) - Append(kappa,split-b) arrived;
[info] 17:14:42.635 [MultiDcSystem-akka.actor.default-dispatcher-17] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,split-b))
[info] 17:14:42.635 [MultiDcSystem-akka.actor.default-dispatcher-17] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965), AppliedSelf(Stored(Append(kappa,split-b)),1509556482635))
[info] 17:14:42.642 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      sending: AllState(List(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965), AppliedSelf(Stored(Append(kappa,split-b)),1509556482635)))
[info] 17:14:44.836 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Append(kappa,split-c) arrived;
[info] 17:14:44.836 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,split-c))
[info] 17:14:44.836 [MultiDcSystem-akka.actor.default-dispatcher-13] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965), AppliedSelf(Stored(Append(kappa,split-b)),1509556482635), AppliedSelf(Stored(Append(kappa,split-c)),1509556484835))
[info] 17:14:44.839 [MultiDcSystem-akka.actor.default-dispatcher-18] INFO  kappa(akka://MultiDcSystem) -      sending: AllState(List(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965), AppliedSelf(Stored(Append(kappa,split-b)),1509556482635), AppliedSelf(Stored(Append(kappa,split-c)),1509556484835)))
[info] 17:17:02.253 [MultiDcSystem-akka.actor.default-dispatcher-2] INFO  kappa(akka://MultiDcSystem) - Append(kappa,after-a) arrived;
[info] 17:17:02.253 [MultiDcSystem-akka.actor.default-dispatcher-2] INFO  kappa(akka://MultiDcSystem) - Applying self event: Stored(Append(kappa,after-a))
[info] 17:17:02.253 [MultiDcSystem-akka.actor.default-dispatcher-2] INFO  kappa(akka://MultiDcSystem) -      State so far: : Vector(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965), AppliedSelf(Stored(Append(kappa,split-b)),1509556482635), AppliedSelf(Stored(Append(kappa,split-c)),1509556484835), AppliedSelf(Stored(Append(kappa,after-a)),1509556622252))
[info] 17:17:02.256 [MultiDcSystem-akka.actor.default-dispatcher-9] INFO  kappa(akka://MultiDcSystem) -      sending: AllState(List(AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,a)),1509555080603), AppliedSelf(Stored(Append(kappa,b)),1509555080603), AppliedSelf(Stored(Append(kappa,c)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080603), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,d)),1509555080604), AppliedSelf(Stored(Append(kappa,e)),1509555088997), AppliedSelf(Stored(Append(kappa,split-a)),1509556478965), AppliedSelf(Stored(Append(kappa,split-b)),1509556482635), AppliedSelf(Stored(Append(kappa,split-c)),1509556484835), AppliedSelf(Stored(Append(kappa,after-a)),1509556622252)))
```


```
$ http 34.240.43.90:8080/introspector/kappa/write/after-a
HTTP/1.1 200 OK
Content-Type: text/plain; charset=UTF-8
Date: Wed, 01 Nov 2017 17:17:02 GMT
Server: akka-http/10.0.10
Transfer-Encoding: chunked

ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,a))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,a))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,b))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,c))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,d))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,d))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,d))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,d))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,d))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,d))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,e))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,split-a))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,split-b))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,split-c))
ReplicatedIntrospector$AppliedSelf::command: Stored(Append(kappa,after-a))
```

though did not heal hm.


Got some bash trouble, not blocker but super confusing:

```
$ lol=nein
ktoso @ 三日月~/code/akka-apps/multidc [wip-on-running-splitting*]
$ if [[ "$lol" -eq "wat" ]]; then 
  echo "yes"
elif [[ "$lol" -eq "nein" ]]; then 
  echo "nope"
fi
yes
```

such problem in network-split.sh -- wat? I mean I'm pretty sure that's ok syntax...

Feel free to pick up this branch! It should be easily followable from the above points (i tried).
Esp. using the rsync + run is very useful to see all logs and poke things.

ktoso commented:
I propose we merge this btw, it's status quo of the infra scripts and worked fine
